scipy/sparse/_bsr.py

```diff
@@ -722,7 +722,10 @@ def isspmatrix_bsr(x):
     return isinstance(x, bsr_matrix) or isinstance(x, bsr_array)
 
 
-class bsr_matrix(spmatrix, bsr_array):
+class bsr_array(_bsr_array, sparray):
```

Comment:
Naive question: must not we use the same method resolution order for sparse arrays and sparse matrices?

Reply:
The order matters for method lookup order. For `bsr_matrix`, presumably there are methods in both classes so the order is important. 

For `bsr_array` the `sparray` class is an empty class (used for it's isinstance impact to indicate an array class). So order doesn't impact the result, and being last might have a very slight speedup for method lookup -- the methods are all in the `_bsr_array` class.

Reply:
Do we want to have the same order (e.g. extend the format mixing first and `sp{array,matrix}` second) for consistency?